src/ArborX_LinearBVH.hpp

```diff
@@ -33,18 +33,19 @@ template <typename DeviceType>
 struct TreeVisualization;
 }
 
-template <typename DeviceType>
+template <typename MemorySpace, typename Enable = void>
 class BoundingVolumeHierarchy
 {
 public:
-  using device_type = DeviceType;
+  using memory_space = MemorySpace;
   using bounding_volume_type = Box;
-  using size_type = typename DeviceType::memory_space::size_type;
+  using size_type = typename MemorySpace::size_type;
 
   BoundingVolumeHierarchy() = default; // build an empty tree
```

Comment:
I would add a `static_assert` for checking that `MemorySpace` is a memory space.

Reply:
Done